mainwindow.cpp

```diff
@@ -871,15 +851,15 @@ void MainWindow::showJohnFinished()
     checkNToggleActionsLastSession();
     // When John stops we need to stop timer and to look status last
     // time.
-    // TODO: currently if john crashed we do not call john.
+    // Currently if john crashed we do not call john.
```

Comment:
On Fri, May 22, 2015 at 11:51:50AM -0700, math07 wrote:
> > @@ -871,15 +851,15 @@ void MainWindow::showJohnFinished()
> >      checkNToggleActionsLastSession();
> >      // When John stops we need to stop timer and to look status last
> >      // time.
> > -    // TODO: currently if john crashed we do not call john.
> > +    // Currently if john crashed we do not call john.
> 
> I don't know. Maybe what Aleksey meant with this TODO is that currently showJohnFinished isn't called when John crashed so callJohnShow() won't be called and the tableview of .pot won't be updated ...

I guess, on Linux you can emulate crash sending SIGSEGV:

$ sleep 1000 &
[1] 1238
$ kill -SEGV 1238
$ 
[1]+  Segmentation fault      sleep 1000

Also kill -9 (sigkill) should work. And all other not catchable
signals.

I don't know what is going when john crashed. I guess it would be
enough to connect our slot of regular stop to signal of abnormal exit.
I am not sure about complications.

Thanks!

-- 
Regards,
Aleksey Cherepanov